src/hypothesis/strategies.py

```diff
@@ -982,32 +987,61 @@ def fractions(min_value=None, max_value=None, max_denominator=None):
 
     If max_denominator is not None then the absolute value of the denominator
     of any generated values is no greater than max_denominator. Note that
-    max_denominator must be at least 1.
+    max_denominator must be None or a positive integer, and the denominator
```

Comment:
Try `fractions(.25, .5, 2).example()` on master - you'll see fairly frequent assertion errors from within `integers()` even though 1/2 is obviously a legal value to return.

Having fixed that, I actually can't determine in the general case whether there are any legal values when the value bounds are at higher precision than allowed for examples.  I'd be very happy to be proved wrong here!